### PR TITLE
Updated documentation to clarify Component's required_packages

### PR DESCRIPTION
**Pull request by hsm207**
**Proposed changes**:
- The list of requirements is actually a list of importable python
modules, not package names. See issue #3647 for more details.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by msamogh** [comment]
Thanks for the PR, we'll give it a review as soon as possible.

File: rasa/nlu/components.py [hunk 1]
```diff
@@ -41,10 +41,12 @@ def validate_requirements(component_names: List[Text]) -> None:
         # if available, use the development file to figure out the correct
         # version numbers for each requirement
         raise Exception(
-            "Not all required packages are installed. "
+            "Not all required importable packages are installed. "
             + "To use this pipeline, you need to install the "
             "missing dependencies. "
-            + "Please install {}".format(", ".join(failed_imports))
+            + "Please install the package(s) that contain the module(s){}".format(
```
Comment (merwok) [inline rasa/nlu/components.py]: I think a space is missing after `module(s)`

Comment (erohmensing) [reply rasa/nlu/components.py]: Good point! Maybe should also include a colon too. Want to make another quick PR since this one was merged already? :)

Comment (hsm207) [reply rasa/nlu/components.py]: I've created PR #4162 to fix this.

**Pull request merged**